Unknown variable "binlog-space-limit" error when using mysql:5.7 image.

Installed the cluster following [this](https://www.presslabs.com/docs/mysql-operator/deploy-mysql-cluster) docs. I am using the `mysql:5.7` image instead of the default `percona` MySQL image. I've also set volumeClaim in `volumeSpec` the cluster yaml, following [this](https://www.presslabs.com/docs/mysql-operator/deploy-mysql-cluster/#persistent-volume-claim). But using this combination of image & own volumeClaim results in the following error:

```
2020-08-24 21:15:24+00:00 [Note] [Entrypoint]: Entrypoint script for MySQL Server 5.7.31-1debian10 started.
2020-08-24 21:15:24+00:00 [ERROR] [Entrypoint]: mysqld failed while attempting to check config
	command was: mysqld --verbose --help
	2020-08-24T21:15:24.711860Z 0 [ERROR] unknown variable 'binlog-space-limit=341M'
2020-08-24T21:15:24.717024Z 0 [ERROR] Aborting
```

I tracked the issue to the code here https://github.com/presslabs/mysql-operator/blob/288131ec062a1165c51562073b28443dbf59f26f/pkg/internal/mysqlcluster/defaults.go#L99

The variable "binlog-space-limit" does not exist for MySQL 5.7 config. Can this check on volumeClaim to set the variable "binlog-space-limit" be restricted to percona images only?

Operator version: 0.4.0

--- a/Changelog.md
+++ b/Changelog.md
@@ -13,6 +13,7 @@ adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
  * Add Google Drive via service account as backup option.
  * Add `initBucketURL` and `initBucketSecretName` options to MysqlCluster chart. This bumps the chart version to `0.3.0`
 ### Changed
+ * Only add `binlog-space-limit` for `percona` image
  * Set timeout of 15s on connection between the operator and Orchestrator
  * Bump controller-util dependency to 0.1.18 which fixes some updates on pod spec.
  * Removed `NO_AUTO_VALUE_ON_ZERO` from `sql-mode` to be inline with MySQL default value

--- a/pkg/internal/mysqlcluster/defaults.go
+++ b/pkg/internal/mysqlcluster/defaults.go
@@ -19,6 +19,7 @@ package mysqlcluster
 import (
 	"fmt"
 	"math"
+	"regexp"
 
 	core "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
@@ -105,8 +106,10 @@ func (cluster *MysqlCluster) SetDefaults(opt *options.Options) {
 				maxBinlogSize = min(binlogSpaceLimit/3, 1*gb)
 			}
 
-			// binlog-space-limit = totalSpace / 2
-			setConfigIfNotSet(cluster.Spec.MysqlConf, "binlog-space-limit", humanizeSize(binlogSpaceLimit))
+			if matched, err := regexp.MatchString("([a-z.]+/)*percona(:.+)?", cluster.Spec.Image); err != nil && matched {
+				// binlog-space-limit = totalSpace / 2
+				setConfigIfNotSet(cluster.Spec.MysqlConf, "binlog-space-limit", humanizeSize(binlogSpaceLimit))
+			}
 
 			// max-binlog-size = min(binlog-space-limit / 4, 1*gb)
 			setConfigIfNotSet(cluster.Spec.MysqlConf, "max-binlog-size", humanizeSize(maxBinlogSize))

--- a/pkg/internal/mysqlcluster/defaults.go
+++ b/pkg/internal/mysqlcluster/defaults.go
@@ -19,7 +19,6 @@ package mysqlcluster
 import (
 	"fmt"
 	"math"
-	"regexp"
 
 	core "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
@@ -106,7 +105,7 @@ func (cluster *MysqlCluster) SetDefaults(opt *options.Options) {
 				maxBinlogSize = min(binlogSpaceLimit/3, 1*gb)
 			}
 
-			if matched, err := regexp.MatchString("([a-z.]+/)*percona(:.+)?", cluster.Spec.Image); err != nil && matched {
+			if cluster.IsPerconaImage() {
 				// binlog-space-limit = totalSpace / 2
 				setConfigIfNotSet(cluster.Spec.MysqlConf, "binlog-space-limit", humanizeSize(binlogSpaceLimit))
 			}

--- a/pkg/internal/mysqlcluster/mysqlcluster.go
+++ b/pkg/internal/mysqlcluster/mysqlcluster.go
@@ -220,11 +220,16 @@ func (c *MysqlCluster) UpdateSpec() {
 	}
 }
 
+// IsPerconaImage checks the MySQL image is percona or not
+func (c *MysqlCluster) IsPerconaImage() bool {
+	return strings.Contains(c.GetMysqlImage(), "percona")
+}
+
 // ShouldHaveInitContainerForMysql checks the MySQL version and returns true or false if the docker image supports or not init only
 func (c *MysqlCluster) ShouldHaveInitContainerForMysql() bool {
 	expectedRange := semver.MustParseRange(">=5.7.26 <8.0.0 || >=8.0.15")
 
-	return strings.Contains(c.GetMysqlImage(), "percona") && expectedRange(c.GetMySQLSemVer())
+	return c.IsPerconaImage() && expectedRange(c.GetMySQLSemVer())
 }
 
 // String returns the cluster name and namespace

